moirae/estimators/online/filters/conversions.py

```diff
@@ -168,8 +175,111 @@ def inverse_transform_samples(self, transformed_samples: np.ndarray) -> np.ndarr
+    the transformation at the pivot point :math:`p`, and :math:`J` is the Jacobian matrix at the pivot. Based on this,
+    the covariance of the transformed vector :math:`f` can be simply expressed as
+    :math:`{\\Sigma}_f = J {\\Sigma}_X J^T`, exactly like that in the
+    :class:`~moirae.estimators.online.filters.conversions.LinearConversionOperator`
+    """
+    @abstractmethod
+    def get_jacobian(self, pivot: np.ndarray) -> np.ndarray:
+        """
+        Method to calculate the Jacobian matrix of the transformation at specified pivot point
+
+        Args:
+            pivot: point to be used as the pivot of the Taylor expansion
+
+        Returns:
+            jacobian: Jacobian matrix
+        """
+        raise NotImplementedError('Implement in child class!')
+
+    @abstractmethod
+    def get_inverse_jacobian(self, transformed_pivot: np.ndarray) -> np.ndarray:
+        """
+        Method to compute the Jacobian matrix of the inverse transformation.
+
+        Args:
+            transformed_pivot: point to be used as the transformed pivot
+
+        Returns:
+            inverse_jacobian: Jacobian matrix of the reverse transformation
+        """
+        raise NotImplementedError('Implement in child function!')
+
+    def transform_covariance(self, covariance: np.ndarray, pivot: np.ndarray) -> np.ndarray:
+        jacobian = self.get_jacobian(pivot=pivot)
+        # Invoke machinery from linear transform
+        linear = LinearConversionOperator(multiplicative_array=jacobian)
+        return linear.transform_covariance(covariance=covariance)
+
+    def inverse_transform_covariance(self,
+                                     transformed_covariance: np.ndarray,
+                                     transformed_pivot: np.ndarray) -> np.ndarray:
+        inv_jacobian = self.get_inverse_jacobian(transformed_pivot=transformed_pivot)
+        # Invoke machinery from linear transform
+        linear = LinearConversionOperator(multiplicative_array=inv_jacobian)
+        return linear.transform_covariance(covariance=transformed_covariance)
+
+
+class AbsoluteValueConversionOperator(FirstOrderTaylorConversionOperator):
+    """
+    Class that performs an absolute value transformation to the samples.
+
+    This is particularly relevant in cases where the estimators used treat values as unbounded, but the underlying
+    models (such as an equivalent circuit) only accept positive parameters. Since the absolute value function is not
+    invertible, the following choice was made for its inverse operations: the inverted values always belong to the
+    positive quadrant of space, that is, they are always positive.
+
+    Args:
+        indices: indices to which the absolute value transformation should be applied; if not provided, applies absolute
+            value to all values
+    """
+    indices: Optional[List[int]] = Field(default=None, description='indices to apply transformation')
```

Comment:
Nice! I like that you thought of only operating on some of the variables.

Reply:
It is critical for the joint estimation, which I tested separately. So I figured it would be useful :D